.pre-commit-config.yaml

```diff
@@ -90,35 +90,37 @@ repos:
         files: tests
         args:
           - --config=.bandit.yml
-  # Run bandit everything but tests directory
+  # Run bandit on everything except the "tests" tree
   - repo: https://github.com/PyCQA/bandit
-    rev: 1.7.0
+    rev: 1.7.4
     hooks:
       - id: bandit
         name: bandit (everything else)
         exclude: tests
   - repo: https://github.com/psf/black
-    rev: 22.3.0
+    rev: 22.12.0
     hooks:
       - id: black
-  - repo: https://gitlab.com/pycqa/flake8
-    rev: 3.9.2
+  - repo: https://github.com/PyCQA/flake8
+    rev: 6.0.0
     hooks:
       - id: flake8
         additional_dependencies:
           - flake8-docstrings
   - repo: https://github.com/PyCQA/isort
-    rev: 5.10.1
+    rev: 5.12.0
     hooks:
       - id: isort
   - repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v0.942
+    rev: v0.991
     hooks:
       - id: mypy
         additional_dependencies:
+          - types-docopt
           - types-requests
+          - types-urllib3
```

Comment:
Should these dependencies be added to `requirements-dev.txt`?  I think @dav3r recently told me that you and he had decided to do that.

If I'm wrong, then I'd like to get some clarification from @dav3r as to what he told me; it seems important.

Reply:
I believe we have only been doing that in Python projects. Currently `mypy` is not included in the `requirements-dev.txt` for Docker projects.

Reply:
It likely stems from the fact that Docker projects will not always (and often don't) include Python code in their configuration. Developing a Docker project that uses Python vs a Python project is another key difference.